src/lib/core/clock_lowres.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2010-2022, Tarantool AUTHORS: please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+#include <signal.h>
+#include <string.h>
+#include <say.h>
+#include <math.h>
+#include "clock.h"
+
+/**
+ * Granularity in seconds.
+ * SIGALRM handler interrupts some syscalls (open, read, etc.), so
+ * too small granularity is not recommended.
+ */
+#define LOW_RES_GRANULARITY 0.05
+
+double low_res_clock = 0.0;
+
+/** A tick of low_res_clock, SIGALRM handler. */
+static void
+clock_lowres_tick(int signum)
+{
+	(void)signum;
+	low_res_clock += LOW_RES_GRANULARITY;
```

Comment:
Addition will accumulate an error. You should reset the value to `clock_monotonic()` on each signal invocation.

Reply:
I don't think we are on the same page: what's the point of this patch then, if we could simply use `clock_gettime`?
https://github.com/tarantool/tarantool/blob/dfa0ea2b4436fab91318f7fb0787db5f9f6f1f75/src/lib/core/clock.c#L42-L48

Reply:
The point is to call `clock_gettime` once X ms.

Reply:
done